cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/hvdc/LossFactor.java

```diff
@@ -0,0 +1,120 @@
+        } else {
+            this.lossFactor1 = 0.0;
+            this.lossFactor2 = 0.0;
+        }
+        // else (i.e. pAC1 == 0 && pAC2 == 0) do nothing: loss factors are null and
+        // stations are probably disconnected
+    }
+
+    private void computeLossFactor1(double pAC1, double poleLossP1, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER)) { // pAC1 > 0
+            this.lossFactor1 = (poleLossP1 / pAC1) * 100;
+        } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && Math.abs(pAC1) + poleLossP1 != 0) { // pAC1 < 0
+            this.lossFactor1 = (poleLossP1 / (Math.abs(pAC1) + poleLossP1)) * 100;
+        }
+    }
+
+    private void computeLossFactor2(double pAC2, double poleLossP2, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER)) { // pAC2 > 0
+            this.lossFactor2 = (poleLossP2 / pAC2) * 100;
+        } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && Math.abs(pAC2) + poleLossP2 != 0) { // pAC2 < 0
+            this.lossFactor2 = (poleLossP2 / (Math.abs(pAC2) + poleLossP2)) * 100;
+        }
+    }
+
+    private void computeLossFactor1FromPAC2(double pAC2, double poleLossP1, double poleLossP2, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && (Math.abs(pAC2) + poleLossP2 + poleLossP1) != 0) { // pAC2 < 0
+            // lossFactor1 = poleLossP1 / pAC1 * 100
+            // pAC1 = pDC + poleLossP1 = pAC2 + poleLossP2 + poleLossP1
+            this.lossFactor1 = (poleLossP1 / (Math.abs(pAC2) + poleLossP2 + poleLossP1)) * 100;
+        } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && (pAC2 - poleLossP2) != 0) { // pAC2 > 0
+            // lossFactor1 = poleLossP1 / pDC * 100
+            // pDC = pAC2 - poleLossP2
+            this.lossFactor1 = (poleLossP1 / (pAC2 - poleLossP2)) * 100;
+        }
+    }
+
+    private void computeLossFactor2FromPAC1(double pAC1, double poleLossP1, double poleLossP2, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && (pAC1 - poleLossP1) != 0) { // pAC1 > 0
+            // lossFactor2 = poleLossP2 / pDC * 100
+            // pDC = pAC1 - poleLossP1
+            this.lossFactor2 = (poleLossP2 / (pAC1 - poleLossP1)) * 100;
+        } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && (Math.abs(pAC1) + poleLossP1 + poleLossP2) != 0) { // pAC1 < 0
+            // lossFactor2 = poleLossP2 / (pDC + poleLossP2) * 100
+            // pDC = pAC1 + poleLossP1
+            this.lossFactor2 = (poleLossP2 / (Math.abs(pAC1) + poleLossP1 + poleLossP2)) * 100;
+        }
+    }
+
+    boolean isOk() {
+        return !Double.isNaN(this.lossFactor1) && !Double.isNaN(this.lossFactor2);
```

Comment:
What happens if only one is not `NaN`? Should we drop its result in any case?

Reply:
Deleted. Loss Factor is always calculated. If lossFactor is NaN it is fixed to 0.0